bayesian/utility/db_gateway.py

```diff
@@ -98,6 +108,80 @@ def get_batch_ca_data(cls, ecosystem: str, packages) -> dict:
         logger.info("It took %s to fetch results from Gremlin.", elapsed_time)
         return response.json()
 
+    @classmethod
+    def get_batch_ca_data_for_pseudo_version(cls, ecosystem: str, packages) -> dict:
+        """Component analyses batch call only for pseudo version applicable for golang."""
+        logger.debug('Executing get_batch_ca_data_for_pseudo_version')
+
+        filter_packages = set()
+        package_version_map = {}
+        for pckg in packages:
+            package_name = pckg['name'].split('@')[0]
+            filter_packages.append(package_name)
+            package_version_map[package_name] = pckg["version"]
+
+        payload = {
+            'gremlin': cls.ca_batch_vuln_query,
+            'bindings': {
+                'ecosystem': ecosystem,
+                'packages': list(filter_packages)
+            }
+        }
+        started_at = time.time()
+        vuln_response = post(url=gremlin_url, data=json.dumps(payload))
+        vuln_response.raise_for_status()
+        elapsed_time = time.time() - started_at
+        logger.info("It took %s to fetch vuln results from Gremlin.", elapsed_time)
+
+        vulnerabilities = {}
+        gh = GithubUtils()
+        for vuln in vuln_response.get('result', {}).get('data', []):
+            package_name = vuln['package_name'][0]
+            if gh._is_commit_date_in_vuln_range(
+               GraphAnalyses.extract_timestamp(
+                   package_version_map[package_name]), vuln['vuln_commit_date_rules'][0]):
+                if package_name not in vulnerabilities:
+                    vulnerabilities[package_name] = []
+                vulnerabilities[package_name].append(vuln)
+
+        payload = {
+            'gremlin': cls.ca_batch_package_query,
+            'bindings': {
+                'ecosystem': ecosystem,
+                'packages': list(vulnerabilities.keys())
+            }
+        }
+        started_pckg_at = time.time()
+        pckg_response = post(url=gremlin_url, data=json.dumps(payload))
```

Comment:
try catch needed here

Reply:
next statement, raise_for_status() takes care of HTTP exception of 4XX and 5XX errors.